### PR TITLE
IM useForm

### PR DESCRIPTION
**Pull request by Perfoys**
Created and used useForm hook

### CODE REVIEW

File: src/components/custom/Input/input.module.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+.input {
+  display: block;
+  font-size: $input-size;
+  height: $input-height;
+  border-radius: $border-radius;
+}
+
+.error {
+  color: red;
```
[the remark this answers is not in the document]
Comment (Perfoys) [reply src/components/custom/Input/input.module.scss:11]: Indeed, my mistake, sorry
Created constant for error-color

File: src/components/modal/Comments/index.tsx [hunk 2]
```diff
@@ -18,23 +20,32 @@ type Inputs = {
...
+        <Form handleSubmit={handleSubmit}>
+          <Input
+            styleClass={style.commentInput}
+            inputType='text'
+            inputName='text'
+            placeholder={t('modal.commentLeave')}
+            handleChange={handleChange}
+            value={fields.text}
+          />
           <label>{t('modal.commentsLabel')}</label>
```
Comment (Smolyaninov-Dmitriy) [inline src/components/modal/Comments/index.tsx:47]: can we add this label to the input component?

File: src/components/custom/Input/index.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const Input: FC<InputProps> = ({ styleClass, inputType, inputName, placeholder, value, handleChange }) => {
+  const [state, setState] = useState({
+    isValid: true,
+    isFocus: false,
+    error: ''
+  });
+  const showError = state.isFocus && !state.isValid;
+
+  const handleValidation = useCallback(() => {
```
Comment (Smolyaninov-Dmitriy) [inline src/components/custom/Input/index.tsx:22]: Let's move this logic in custom hook. And will be call it before submit. What do u think about it?

File: src/components/custom/Form/form.test.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { shallow } from 'enzyme';
+import Form from '.';
+
+describe('Form component testing', () => {
+  const handleSubmit = jest.fn();
+  test('Form component renders properly', () => {
+    const wrapper = shallow(<Form handleSubmit={handleSubmit} ><h1>test</h1></Form>);
```
[the remark this answers is not in the document]
Comment (Perfoys) [reply src/components/custom/Form/form.test.tsx:7]: Yes, solved this problem

File: src/components/modal/Comments/index.tsx (hunk 2 above)
Comment (Perfoys) [reply src/components/modal/Comments/index.tsx:47]: Yes, moved label to the input component

File: src/components/modal/Comments/index.tsx [hunk 5]
```diff
@@ -17,24 +19,35 @@ type Inputs = {
 };
 
 const Comments: FC<CommentsProps> = ({ task }) => {
+  const inputName = 'text';
```
Comment (Smolyaninov-Dmitriy) [inline src/components/modal/Comments/index.tsx:22]: these variables can be moved outside the component. usually uppercase letters are used for such variables

File: src/components/modal/Comments/index.tsx [hunk 6]
```diff
@@ -17,24 +19,35 @@ type Inputs = {
...
+  const inputType = 'text';
   const { t } = useTranslation();
-  const { register, handleSubmit, reset } = useForm<Inputs>({ defaultValues: { taskId: task.id, author: task.member } });
+  const { values, handleChange, handleSubmit, handleReset, errorMessage } = useForm({ text: { type: inputType, value: '' } });
```
Comment (Smolyaninov-Dmitriy) [inline src/components/modal/Comments/index.tsx:25]: could we use variable instead 'text' inside useForm?
`useForm({ [INPUT_NAME]: { type: inputType, value: '' } })`

File: src/hooks/useForm.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  const handleReset = useCallback(() => {
+    setValues(defaultValues);
+  }, []);
+
+  const handleValidation = (data: FormProps): boolean => {
+    const currentErrors: Error = {};
+    const fields = Object.entries(data);
+
+    fields.forEach((field: [string, Field]) => {
+      const key = field[0];
```
Comment (Smolyaninov-Dmitriy) [inline src/hooks/useForm.tsx:39]: Can we use destructuring instead of indexes?

Comment (Perfoys) [reply src/hooks/useForm.tsx:39]: Yes, we can, it will be better

File: src/components/modal/Comments/index.tsx (hunk 5 above)
Comment (Perfoys) [reply src/components/modal/Comments/index.tsx:22]: My bad, fixed it

File: src/components/modal/Comments/index.tsx (hunk 6 above)
Comment (Perfoys) [reply src/components/modal/Comments/index.tsx:25]: Yes, solved this problem, thank you

**Pull request merged**